### PR TITLE
ci: added schedule event name in package-release and master steps

### PR DESCRIPTION
**Pull request by sharanya-appsmith**
#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

- [ ] Manual

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Chores**
	- Updated workflow to trigger certain job steps on a scheduled basis.

### CODE REVIEW

**Comment by sharanya-appsmith** [comment]
TBP Run: https://github.com/appsmithorg/appsmith/actions/runs/7284501745

File: .github/workflows/test-build-docker-image.yml [hunk 1]
```diff
@@ -283,7 +284,7 @@ jobs:
 
     steps:
       - name: Checkout the head commit of the branch
-        if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
+        if: github.event_name == 'push' || github.event_name == 'workflow_dispatch' || github.event_name == 'schedule'
```
Comment (sarojsarab) [inline .github/workflows/test-build-docker-image.yml:287]: Why you need schedule for master? when the workflow runs with cron it always fetches the base branch in this case the base branch is release, so I don't think we need this here.

**Review by sarojsarab — CHANGES_REQUESTED** [review]

File: .github/workflows/test-build-docker-image.yml (hunk 1 above)
Comment (sharanya-appsmith) [reply .github/workflows/test-build-docker-image.yml:287]: Doesn't package-master step need this? Cos its skipped here? 
Screenshot 2023-12-21 at 8 03 53 PM

Comment (sarojsarab) [reply .github/workflows/test-build-docker-image.yml:287]: Why you need to package the master branch when you are building for release branch?

Comment (sharanya-appsmith) [reply .github/workflows/test-build-docker-image.yml]: Removed.

**Review by sarojsarab — APPROVED** [review]

**Pull request merged**